lib/util/http/auth.ts

```diff
@@ -7,6 +7,14 @@ import {
 } from '../../constants/platforms';
 import { GotOptions } from './types';
 
+// Using the datasource constants would result into a circular dependency
+const GITHUB_HOST_TYPES = [
+  PLATFORM_TYPE_GITHUB,
+  'github-releases',
+  'github-tags',
+  'pod',
```

Comment:
I don't like this change, makes it pretty complicated to get all those types in sync.

Please move this to `lib/types/github.ts`. We should make some typescript assertions, so that this works if anything is changes. @JamieMagee Do you have an idea how to get this properly typed?

Reply:
Yes! Also don't like it! Could create `lib/types/github.ts` but what we need to ensure is that the datasource ids and the list of github host types don't diverge. 👍🏻

Reply:
Hey @JamieMagee, do you have any thoughts? Otherwise, I would do the suggested changes of @viceice

Reply:
I can't really think of a better way of doing this. It doesn't make sense, logically, for anything other than the datasource itself to own its own type/id.